File: flockScreen.cpp
/*
 * flockScreen.cpp
 *
 *  Created on: 20 Nov 2017
 *      Author: Abdelrahman Barghouth
 */

#include <iostream>
#include <string>
#include <math.h>
#include <unistd.h>

using namespace std;

int main(int argc, char** argv)
{
	string str1,str2;
//	str1 = argv[1];
//	str2 = argv[2];

	cout << "Please enter the first string" << endl;
	getline(cin,str1);
	cout << str1<<endl;

	cout << "Please enter the second string" << endl;
	getline(cin,str2);


	int depth = -1, result =0, stringLength = str1.length();

	if(str1.length()!=str2.length())
	{
		cerr << "Error: Strings are not of the same length" << endl;
		return -1;
	}

	for(int i=0; i<stringLength;i++)
	{
		if(str1[i] == str2[i])
		{
			if(str1[i] == '[')
				depth++;
			else if(str1[i] == ']')
				depth--;
		}
		else if(isalpha(str1[i]) && isalpha(str2[i]))
			result+=pow(2,depth);
		else
		{
			cerr << "Error: Strings are not matched either in depth or fragment size" << endl;
			return -1;
		}
	}
	cout << "Say the magic words and the result will appear!!" << endl;
	sleep(2);
	cout << "Well done!! The result is = " << result << endl;
	return 0;
}

File: flockScreen.py
#
# flockScreen.py
# Author: Abdelrahman Barghouth
# Date: 18/11/2017
#

import math
import time
from sys import stdin
import sys

def calculateResult(str1,str2):
	depth = -1
	result =0

	stringLength = len(str1)

	if(len(str1)!=len(str2)):
		sys.exit("Error: Strings are not of the same length")

	for i in range (0,stringLength):
		if(str1[i] == str2[i]):
			if(str1[i] == '['):
				depth=depth+1
			elif(str1[i] == ']'):
				depth=depth-1
		elif(str1[i].isalpha==1 and str2[i].isalpha==1):
			result=result+pow(2,depth)
		else:
			sys.exit("Error: Strings are not matched either in depth or fragment size")
	return result

print("Please enter the first string")
str1 = stdin.readline()

print("Please enter the second string")
str2 = stdin.readline()

result = calculateResult(str1,str2)

print( "Say the magic words and the result will appear!!")
time.sleep(2)
print("Well done!! The result is = ", result)
